zebra-chain/src/block/tests.rs

```diff
@@ -110,63 +107,60 @@ fn blockheaderhash_from_blockheader() {
 #[test]
 fn deserialize_blockheader() {
     // https://explorer.zcha.in/blocks/415000
-    let _header =
-        BlockHeader::zcash_deserialize(&zebra_test::vectors::HEADER_MAINNET_415000_BYTES[..])
-            .expect("blockheader test vector should deserialize");
+    let _header = zebra_test::vectors::HEADER_MAINNET_415000_BYTES
+        .zcash_deserialize_into::<BlockHeader>()
+        .expect("blockheader test vector should deserialize");
 }
 
 #[test]
 fn deserialize_block() {
-    Block::zcash_deserialize(&zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES[..])
+    zebra_test::vectors::BLOCK_MAINNET_GENESIS_BYTES
```

Comment:
This is actually black magic and I have no idea why this works without doing `&ident[..]` first on the new version but doesn't work with the old version. if anyone else understands this pls explain it to me

Reply:
Looks like automatic referencing and dereferencing for `self`:
https://doc.rust-lang.org/stable/book/ch05-03-method-syntax.html#wheres-the---operator

Reply:
oh shit I never realized it only applies to self, huge win imo